cmake/kokkos_arch.cmake

```diff
@@ -802,6 +802,14 @@ IF(KOKKOS_ARCH_INTEL_PVC)
   CHECK_MULTIPLE_INTEL_ARCH()
 ENDIF()
 
+IF (KOKKOS_ENABLE_OPENMP)
+  IF(KOKKOS_CLANG_IS_CRAY)
+    COMPILER_SPECIFIC_LINK_OPTIONS(
+      Clang -fopenmp
```

Comment:
Did you try this?
```suggestion
      Clang -fopenmp
      CrayClang -fopenmp
```

Reply:
It looks like [this](https://github.com/kokkos/kokkos/blob/develop/cmake/kokkos_compiler_id.cmake#L96) condition is not satisfied for `CrayClang` with cmake>3.28. So we might have to fix Kokkos identification as @crtrott suggested.

Reply:
You will likely also need to adjust https://github.com/kokkos/kokkos/blob/af8df138190527f4a99b632da0a58faec188e8ba/cmake/kokkos_functions.cmake#L849.